find_new/src/find_new.py

```diff
@@ -72,6 +74,32 @@ def add_invalid_product_record(event_id, granule, message):
     database.put_product(product)
 
 
+def get_neighbors(granule_name: str, max_neighbors: int = 2) -> List[dict]:
+    results = asf_search.product_search([granule_name])
+    assert len(results) == 1
+    granule: asf_search.ASFProduct = results[0]
+
+    stack = asf_search.baseline_search.stack_from_product(granule)
+    stack = [item for item in stack if item.properties['temporalBaseline'] < 0]
+    neighbors = [item.properties['fileID'] for item in stack[-max_neighbors:]]
+
+    response = requests.post(
+        SEARCH_URL,
+        params={
+            'product_list': ','.join(neighbors),
+            'output': 'jsonlite'
+        }
+    )
+
+    status_code = str(response.status_code)
+    if status_code[0] == '4':
+        raise asf_search.ASFSearch4xxError()
+    elif status_code[0] == '5':
+        raise asf_search.ASFSearch5xxError()
+
+    return response.json()['results']
```

Comment:
This final search is just to get `jsonlite` output from this function, correct?

I think you could avoid this by returning just the neighbor names and doing:
```python
    for neighbor_name in neighbor_namess:
        insar_job = hyp3.prepare_insar_job(
            granule['granuleName'], neighbor_name], include_look_vectors=True, apply_water_mask=True
        )
```

or updating this line:
https://github.com/ASFHyP3/hyp3-event-monitoring/blob/08f8267d8c0416d3c6df4584d58d1973b357ba53/find_new/src/find_new.py#L123

to
```
granule['granuleName'], neighbor['sceneName'], include_look_vectors=True, apply_water_mask=True
```

since that's the only place `neighbors` appears to get used. 

It *is* a little awkward to have `granule['granuleName'], neighbor_name` or `granule['sceneName'], neighbor['sceneName']` which might be worth the extra search. 

I think I'll defer to you on this one as you've probably done more thinking about this than I have.

Reply:
I went down a similar train of thought, but each `neighbor` is also appended to `granule_lists` [here](https://github.com/ASFHyP3/hyp3-event-monitoring/blob/08f8267d8c0416d3c6df4584d58d1973b357ba53/find_new/src/find_new.py#L126) and then each granule is formatted [here](https://github.com/ASFHyP3/hyp3-event-monitoring/blob/08f8267d8c0416d3c6df4584d58d1973b357ba53/find_new/src/find_new.py#L43) before being put in the database [here](https://github.com/ASFHyP3/hyp3-event-monitoring/blob/08f8267d8c0416d3c6df4584d58d1973b357ba53/find_new/src/find_new.py#L139). Changing the return type for `get_neighbors` would require at least updating `format_product` and `format_granule`, as well as refactoring `test_format_product`, `test_format_granule`, `test_submit_jobs_for_granule`, and `test_lambda_handler`, possibly others. I decided the extra search was worth avoiding the additional refactoring.

Reply:
:+1: